Updating beginner module documentation. (#29)
moja-global/Google.Season.of.Documentation: modules-development/gcbm-module.md

In GCBM, forest inventory and disturbance events are spatially explicit. The location of disturbances are explicit in spatial layers instead of rule based. The GCBM Module is able to simulate large landscapes easily.

![image](assets/gcbm.png)

> It is one of the Country specific modules designed especially for Canada.

Reviewer: ```suggestion
The GCBM module is one of the country specific modules designed especially for Canada.
```

    The CaMP was designed as a module for the spatially-explicit Generic Carbon Budget Model (GCBM) that is intended for future reporting of Greenhouse gas (GHG) emissions and removal estimates from Canada's managed forest area. 

    The Canadian Model for Peatlands (CaMP) is a site- to national-level peatland carbon model developed to better estimate greenhouse gas (GHG) emissions across a range of peatland types within Canada.

    Different events are fired on the basis of different land type found during simulation.
    like If peatland land type is encountered then fires peatland disturbance event else it fires a regular CBM disturbance event.

    For More information regarding CaMP check out this [Document](https://www.sciencedirect.com/science/article/pii/S0304380020302350).

2. **Decay Module**

    This Module performs the annual decay and turnover on a set of dead organic matter pools present in the GCBM.
    Data Requirements for the Decay Module are as :
    
    - A table named "decay_parameters" with one set of decay parameters for each of the enumerated dom pools in the DomPool.

>     - Scalar "mean_annual_temperature" is the mean annual temperature of the environment.

Reviewer: ```suggestion
    - Scalar "mean_annual_temperature". This is the mean annual temperature of the environment.
```

>     - Scalar "SlowMixingRate" the amount turned over from slow aboveground to slow belowground annually.

Reviewer: ```suggestion
    - Scalar "SlowMixingRate". This is the amount that has moved from slow aboveground to slow belowground annually.
```

3. **Moss Decay Module**

    This module is used for moss related computing, that is whether the moss pool is decaying or not.
    
    It uses various mathematical equations to apply decay rates to the different moss pools.

4.  **Moss Disturbance Module**

    This module responds to the fire disturbance events in CBM.
    

>     This module is responsible for transferring carbon content from Moss pools to the Greenhouse gases pools.

Reviewer: ```suggestion
    This module is responsible for transferring carbon content from moss pools to the GHG pools.
```

>     It gets the input data from the variable ```moss_fire_parameters```.

Reviewer: ```suggestion
    It gets the input data from the variable: ```moss_fire_parameters```.
```

5. **Moss Growth Module**

>     This module is responsible for the growth of moss pools that is it increments the moss growth with the help of mathematical equations, various parameters and moss pools.

Reviewer: ```suggestion
    This module is responsible for the growth of moss pools. It increments the moss growth with the help of mathematical equations, various parameters and moss pools.
```

6. **Moss Turnover Module**

    This module is responsible for the turnover of moss pools that is from the moss live pool to the moss past pool.
    
    It accurately calculates the amount of turnover moss by taking moss live amount and transfer rates into consideration and by applying transfer events. 

7. **Peatland after CBM Module**

    This module is triggered after CBM simulation on a land unit, and prepares the land unit to simulate the peatland simulation.

    It transfers carbon from some CBM pools to  the peatland pools. 
    
    It is called after finishing the regular CBM simulation.

8. **Peatland Decay Module**

>     This module is responsible for Peatland decay.

Reviewer: ```suggestion
    This module is responsible for peatland decay.
```

    
    It gets the data by variable ```peatland_decay_parameters```, ```peatland_turnover_parameters``` and also accounts in various parameters like mean annual temperature, annual water table depth, total initial carbon and transfers the pool's content accordingly.

    It also uses a special turnover rate for various pools like carotelm pool.

>     It sets the turnover rate for diffrent pools by taking into acoount of values of varoius greenhouse gases.

Reviewer: ```suggestion
    It sets the turnover rate for different pools by taking into account the values of various GHG.
```

9. **Peatland Disturbance Module**

    This module responds to the historical and last disturbance events in the CBM spinup.
    
    It alters the water table depth according to the disturbance type and Peatland Id. 

10. **Peatland Growth Module**

    This module gets the data from the ```peatland_growth_parameters```, ```peatland_turnover_parameters``` and ```peatland_growth_curve```.
    
    It simulates the various growth cycles, growth curves and also sets the component value for the pools like woody layer, moss layer and many more. 

11. **Tree Growth Module**

    This module deals with the small trees growth curve ID like black spruce.
    
    There is a peatland pre-defined forest growth curve of black spruce.
    
    It takes the current eco boundary variable name and then checks if it was changed or just set and as output, it updates the small tree growth curve parameter.
    

>     It calculates the biomass transfer between various tree pools like stem wood, coarse root, etc with the help of the age of the small tree.

Reviewer: ```suggestion
    It calculates the biomass transfer between various tree pools like stem wood, coarse root, with the help of the age of the small tree.
```
